File: AndroidMVVMExample/app/src/main/java/com/blackbirds/shakil/androidmvvmexample/ViewModel/MainActivityViewModel.java
package com.blackbirds.shakil.androidmvvmexample.ViewModel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainActivityViewModel extends ViewModel {
    MutableLiveData<String> resultMutableLivedata;

    public MutableLiveData<String> addFunction(String numberOne, String numberTwo){
        int numOne = Integer.parseInt(numberOne);
        int numTwo = Integer.parseInt(numberTwo);
        int result = numOne + numTwo;

        if (resultMutableLivedata == null){
            resultMutableLivedata = new MutableLiveData<>();
        }
        resultMutableLivedata.setValue(String.valueOf(result));

        return resultMutableLivedata;
    }
}

File: MVVMExampleKotlin/app/src/main/java/com/blackbirds/shakil/mvvmexamplekotlin/viewModel/MainActivityViewModel.kt
package com.blackbirds.shakil.mvvmexamplekotlin.viewModel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MainActivityViewModel : ViewModel() {
    var resultMutableLivedata: MutableLiveData<String>? = null
    fun addFunction(numberOne: String, numberTwo: String): MutableLiveData<String> {
        val numOne = numberOne.toInt()
        val numTwo = numberTwo.toInt()
        val result = numOne + numTwo
        if (resultMutableLivedata == null) {
            resultMutableLivedata = MutableLiveData()
        }
        resultMutableLivedata!!.value = result.toString()
        return resultMutableLivedata as MutableLiveData<String>
    }
}